### PR TITLE
WIP: rely solely on ophyd metadata subscriptions

### PR DESCRIPTION
**Pull request by klauer**
(WIP, including some other semi-related fixes)

### CODE REVIEW

File: typhon/plugins/core.py [hunk 1]
```diff
@@ -101,13 +104,70 @@ def send_new_value(self, value=None, **kwargs):
...
+            if self.signal_type is not np.ndarray:
+                value = self.signal_type(value)
             self.new_value_signal[self.signal_type].emit(value)
         except Exception:
             logger.exception("Unable to update %r with value %r.",
                              self.signal.name, value)
 
+        if severity is not None:
+            try:
+                self.new_severity_signal.emit.emit(severity)
```
Comment (teddyrendahl) [inline typhon/plugins/core.py]: `emit.emit`?

File: .travis.yml [hunk 2]
```diff
@@ -39,6 +39,8 @@ install:
   - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION typhon --file dev-requirements.txt
   # Launch Conda environment
   - source activate test-environment
+  # temporarily install ophyd master
+  - pip install -U git+https://github.com/NSLS-II/ophyd.git
```
Comment (teddyrendahl) [inline .travis.yml]: Might not be a bad idea just to add `lightsource-dev` to the build matrix permanently 🤷‍♂️

File: typhon/plugins/core.py [hunk 3]
```diff
@@ -101,13 +104,70 @@ def send_new_value(self, value=None, **kwargs):
...
+            info = self.signal.describe()[self.signal.name]
+            # Shove in connected, read_access, write_access into the desc to
+            # make my generic loop simpler, thereby confusing everyone else
+            info.update(**kwargs)
+        else:
+            # We only have legitimate access to what's given in the metadata
+            # callback, as the signal isn't connected yet :(
+            info = kwargs
+
+        key_to_signal = {
```
Comment (teddyrendahl) [inline typhon/plugins/core.py]: I have a feeling this dictionary might be generally useful. Thoughts about putting it as a class attribute?

File: typhon/plugins/core.py [hunk 4]
```diff
@@ -101,13 +104,70 @@ def send_new_value(self, value=None, **kwargs):
...
+            'connected': self.connection_state_signal,
+            'write_access': self.write_access_signal,
+            # 'read_access' unused
+        }
+
+        enum_strs = info.get('enum_strs', None)
+        if enum_strs is not None:
+            info['enum_strs'] = tuple(enum_strs)
+
+        print(self.signal.name, info)
```
Comment (teddyrendahl) [inline typhon/plugins/core.py]: A stray print. Just putting here as a reminder.

File: typhon/signal.py [hunk 5]
```diff
@@ -43,18 +43,9 @@ def signal_widget(signal, read_only=False):
         PyDMLabel, PyDMLineEdit, or PyDMEnumComboBox based on whether we should
         be able to write back to the widget and if the signal has ``enum_strs``
     """
-    # Grab our channel name
-    # Still re-route EpicsSignal through the ca:// plugin
-    if isinstance(signal, EpicsSignalBase):
```
Comment (teddyrendahl) [inline typhon/signal.py]: 🎉

File: tests/test_signal.py [hunk 6]
```diff
@@ -6,9 +6,9 @@
 # External #
 ############
 from ophyd import Kind
-from ophyd.signal import Signal, EpicsSignal, EpicsSignalRO
-from ophyd.sim import SynSignal, SynSignalRO
-from ophyd.tests.conftest import using_fake_epics_pv
```
Comment (teddyrendahl) [inline tests/test_signal.py]: 😱 forgot this was here. Nice catch!

**Comment by klauer** [comment]
There's still some upstream ophyd stuff that needs to get in, so we should hold off on this for a bit. Maybe it's time to cherry-pick the other fixes?

I'll try to get those ophyd changes out soon, but it'll take a while to get that reviewed/cleared with everyone else.

**Comment by klauer** [comment]
Pending nsls-ii/ophyd#653
Will benefit from nsls-ii/ophyd#659

**Pull request closed without merging**